cpp/src/Ice/Reference.cpp

```diff
@@ -1206,12 +1192,13 @@ IceInternal::RoutableReference::toProperty(const string& prefix) const
     return properties;
 }
 
-int
-IceInternal::RoutableReference::hashInit() const
+size_t
+IceInternal::RoutableReference::hash() const noexcept
 {
-    int value = Reference::hashInit();
-    hashAdd(value, _adapterId);
```

Comment:
In a remarkable twist, we were not including endpoints at all in the hash. As a result, all the endpoint hashing was completely unused!

I double checked - this code is identical on 3.7.